src/Joomlatools/Console/Command/Extension/Install.php

```diff
@@ -49,17 +53,64 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         parent::execute($input, $output);
 
-        $this->extensions = (array) $input->getArgument('extension');
+        $extensions = (array) $input->getArgument('extension');
+
+        $this->composer_extensions = array_filter(array_map(function($x){
+            if(strpos($x, '/') !== false){
+                return $x;
+            }
+        }, $extensions));
+
+        if(count($this->composer_extensions)) {
+            $extensions = array_diff($extensions, $this->composer_extensions);
+        }
+
+        $this->extensions = $extensions;
 
         $this->check($input, $output);
-        $this->install($input, $output);
+
+        if (count($this->composer_extensions)) {
+            $this->composer_install($input, $output);
+        }
+
+        if (count($this->extensions)) {
+            $this->install($input, $output);
+        }
     }
 
     public function check(InputInterface $input, OutputInterface $output)
     {
         if (!file_exists($this->target_dir)) {
             throw new \RuntimeException(sprintf('Site not found: %s', $this->site));
         }
+
+        if (count($this->composer_extensions))
+        {
+            $result = shell_exec('composer -v > /dev/null 2>&1 || { echo "false"; }');
+
+            if (trim($result) == 'false' && !file_exists($this->target_dir . '/composer.phar'))
+            {
+                $output->writeln('<error>You need composer installed either locally or globally: https://getcomposer.org/doc/00-intro.md</error>');
+                exit();
+            }
+
+            if ($result != 'false'){
+                $this->composer_global = true;
+            }
+        }
+    }
+
+    public function composer_install(InputInterface $input, OutputInterface $output)
+    {
+        chdir($this->target_dir);
```

Comment:
@yiendos It could be easier to just set the `--working-dir` argument when invoking the Composer command, so we don't have to keep track of PHP's working directory. See: https://getcomposer.org/doc/03-cli.md#global-options